test/over_react/component_declaration/flux_component_test.dart

```diff
@@ -65,6 +65,18 @@ void main() {
           reason: 'component should no longer be listening after unmount');
     });
 
+    test('does not subscribe to a store that is disposed or disposing', () async {
+      var store = new TestStore();
+      var renderedInstance = render(TestDefault()..store = store);
+      TestDefaultComponent component = getDartComponent(renderedInstance);
+      expect(component.subscriptions.length, 1);
+
+      await store.dispose();
+      renderedInstance = render(TestDefault()..store = store);
+      component = getDartComponent(renderedInstance);
+      expect(component.subscriptions.length, 0);
```

Comment:
Alternatively, this could test that rendering does not cause an exception (when the store is listened to while disposing).  Technically, that tests the failure case that we're addressing, but it also asserts the expected behavior in a weird way that is dependent on behavior from another repo.  But it also wouldn't require the subscriptions list to be exposed.

:man_shrugging: